### PR TITLE
Modify stuff for Quarkus Native compile and exec

### PR DESCRIPTION
**Pull request by shokakucarrier**
deleted unused and duplicated lib, add resource-config&reflection-config for native compile support; modified interface name & return type, add some annotation to address native execute problem

### CODE REVIEW

**Comment by shokakucarrier** [comment]
@ruhan1 @yma88 , I've tested this patch, we can now compile these codes to native executable file, the config files should be placed in /project/config/proxy.yaml and so.
BTW the precompiled container image can be find in: quay.io/kaine/indy-sidecar:native-latest

File: src/main/java/org/commonjava/util/sidecar/services/ProxyService.java [hunk 1]
```diff
@@ -213,12 +221,12 @@ private String getTraceId( MultiMap headers )
     }
 
     @FunctionalInterface
-    private interface CheckedFunction<T, R>
+    private interface Function<T, R>
```
Comment (shokakucarrier) [inline src/main/java/org/commonjava/util/sidecar/services/ProxyService.java:224]: This is important, not just a name change, only Function<T,R> works in GraalVM compilation

**Comment by ruhan1** [comment]
Looks great! I don't know if there is any change wrt the debug, e.g. java stacktrace output. This is a good start and we can learn along the way.

**Comment by yma88** [comment]
LGTM

**Comment by shokakucarrier** [comment]
> Looks great! I don't know if there is any change wrt the debug, e.g. java stacktrace output. This is a good start and we can learn along the way.

I don't have any change related to debug, I'm using a mac so I can't run the executable directly, and I don't know how to use gdb to debug quarkus code

**Pull request merged**